c10/core/TensorImpl.h

```diff
@@ -1579,7 +1580,13 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
   // This pointer always has unique ownership (meaning only one TensorImpl can own it
   // at a time).
   // This is private because we must maintain dispatcher invariants on it
-  // in type_set_.
+  // in type_set_, namely, that autograd_meta_ != nullptr iff type_set_.has(VariableTensorId).
+  //
+  // NB: We CANNOT default this to nullptr, as that will result in an invocation
+  // of std::default_delete<AutogradMeta> which won't work as AutogradMeta
+  // is incomplete at this point.  But the defaulting to nullptr is also
+  // pointless because unique_ptr has a default constructor which will do the
+  // right thing.
```

Comment:
Is the `NB:` part of the comment meant for another PR? Because it says `We CANNOT default this to nullptr` but we still do in this PR :D

Reply:
Yeah it's wrong. I just fixed it